### PR TITLE
fix brc20 self-mint bugs

### PR DESCRIPTION
**Pull request by MeihaoZuyu**

### CODE REVIEW

File: ord/getter/opiclienttest.go [hunk 1]
```diff
@@ -77,6 +80,7 @@ func NewOPIOrdGetterTest(config *DatabaseConfig, latestBlockHeight uint) (*OPIOr
 			SentAsFee:     bool(sent_as_fee),
 			Content:       []byte(content),
 			ContentType:   content_type,
+			ParentID:  parent_id,
```
Comment (MeihaoZuyu) [inline ord/getter/opiclienttest.go:83]: Seems like you didn't `gofmt` the code? 🤔

**Review by MeihaoZuyu — COMMENTED** [review]
LGTM

File: apis_test.go [hunk 2]
```diff
@@ -133,11 +133,11 @@ func loadVerifyCurrentBalanceOfPkscript(tick string, pkScript string, catchupHei
...
 
 func TestAPI_VerifyCurrentBalanceOfWallet(t *testing.T) {
-	loadVerifyCurrentBalanceOfWallet("meme", "bc1prvqdfjku8359hk9uc2tdgg0xlwvsel2fjr9ysydmaas9x3kyzuvskuwmlq", uint(779980), t)
+	loadVerifyCurrentBalanceOfWallet("meme", "bc1prvqdfjku8359hk9uc2tdgg0xlwvsel2fjr9ysydmaas9x3kyzuvskuwmlq", uint(779980), t, 782000)
 }
 
-func loadVerifyCurrentBalanceOfWallet(tick string, wallet string, catchupHeight uint, t *testing.T) {
-	ordGetterTest, arguments := loadMain()
+func loadVerifyCurrentBalanceOfWallet(tick string, wallet string, catchupHeight uint, t *testing.T, loadFile uint) {
+	ordGetterTest, arguments := loadMain(loadFile)
```
Comment (archidoge0) [inline apis_test.go:140]: `loadFile` -> `loadHeight`

File: opi_compare_test.go [hunk 3]
```diff
@@ -16,13 +16,13 @@ func Test_OPI(t *testing.T) {
...
 		log.Fatalf(fmt.Sprintf("error happened: %v", err))
 	}
-	ordGetterTest, arguments := loadMain()
+	ordGetterTest, arguments := loadMain(782000)
 	queue, err := CatchupStage(ordGetterTest, &arguments, stateless.BRC20StartHeight-1, latestHeight)
 	if err != nil {
 		log.Fatalf(fmt.Sprintf("error happened: %v", err))
 	}
 	ordGetterTest.LatestBlockHeight = latestHeight
-	go ServiceStage(ordGetterTest, &arguments, queue, 10*time.Millisecond)
```
Comment (archidoge0) [inline opi_compare_test.go:25]: No need to change 50ms -> 10 ms.

File: apis_test.go [hunk 4]
```diff
@@ -133,11 +133,11 @@ func loadVerifyCurrentBalanceOfPkscript(tick string, pkScript string, catchupHei
 }
 
 func TestAPI_VerifyCurrentBalanceOfWallet(t *testing.T) {
-	loadVerifyCurrentBalanceOfWallet("meme", "bc1prvqdfjku8359hk9uc2tdgg0xlwvsel2fjr9ysydmaas9x3kyzuvskuwmlq", uint(779980), t)
+	loadVerifyCurrentBalanceOfWallet("meme", "bc1prvqdfjku8359hk9uc2tdgg0xlwvsel2fjr9ysydmaas9x3kyzuvskuwmlq", uint(779980), t, 782000)
 }
 
-func loadVerifyCurrentBalanceOfWallet(tick string, wallet string, catchupHeight uint, t *testing.T) {
-	ordGetterTest, arguments := loadMain()
+func loadVerifyCurrentBalanceOfWallet(tick string, wallet string, catchupHeight uint, t *testing.T, loadFile uint) {
```
Comment (archidoge0) [inline apis_test.go:139]: loadFile -> loadHeight

File: selfMint_test.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+	"github.com/RiemaLabs/modular-indexer-committee/ord"
+	"github.com/RiemaLabs/modular-indexer-committee/ord/stateless"
+)
+
+func Test_SelfMint(t *testing.T) {
+	stateless.SelfMintEnableHeight = 779832
+	var latestHeight uint = stateless.BRC20StartHeight + ord.BitcoinConfirmations
+	loadVerifyCurrentBalanceOfWallet("xordi", "bc1pkj5jjzglh99zxqu6w9vwdlpk7rqr706jw8t2jtsf4yvfrrvc6ggqlefhke", latestHeight, t, 779838)
+	// loadVerifyCurrentBalanceOfWallet("ordi", "bc1pxaneaf3w4d27hl2y93fuft2xk6m4u3wc4rafevc6slgd7f5tq2dqyfgy06", latestHeight, t)
```
Comment (archidoge0) [inline selfMint_test.go:14]: Remove this line.

File: ord/getter/opiclienttest.go (hunk 1 above)
Comment (archidoge0) [reply ord/getter/opiclienttest.go:83]: @AugustNorden Is it convenient to add a `gofmt` check in CICD? cc @MeihaoZuyu

Comment (MeihaoZuyu) [reply ord/getter/opiclienttest.go]:
```bash
goimports -w .
```

This tool formats and optimize imports for the whole workspace.

**Review by archidoge0 — APPROVED** [review]
LGTM

**Pull request merged**